backend/dataall/modules/loader.py

```diff
@@ -0,0 +1,64 @@
+"""Load modules that are specified in the configuration file"""
+import importlib
+import logging
+from enum import Enum
+from typing import List
+
+from dataall.core.config import config
+
+log = logging.getLogger(__name__)
+
+_MODULE_PREFIX = "dataall.modules"
+
+
+class ImportMode(Enum):
+    """Defines importing mode
+
+    Since there are different infrastructure components that requires only part
+    of functionality to be loaded, there should be different loading modes
+
+    Keys represent loading mode while value a suffix im module loading.
+    For example, API will try to load a graphql functionality under a module directory
```

Comment:
Can modules be self-contained and import all infra components they require to properly function themselves e.g. in the modules `__init__.py`? Or there is a use case where user might only want to import APIs from the module but not the tasks for example (given that if there's both in the module, the APIs probably launch tasks)?

Reply:
That was my original approach, but it crashed on the rocks of reality 🙂 
Whenever we want to load all submodules of a module, the submodules will start the registration into the application.
In our infrastructure, we have three components that uses different modes: GraphQL Lambda, Lambda for async work, and ECS for deployment.  A user might want to load API in GraphQL Lambda and the lambda will launch the tasks, but it will delegate the run to ECS which was loaded with the CDK mode. 

When we load the whole module in ECS as an example, gql part will try to register it in ariadne framework and will crash.